### PR TITLE
Refactor player selection implementation and command line formatter

### PR DESCRIPTION
**Pull request by jonmanahan**

### CODE REVIEW

**Comment by jonmanahan** [comment]
> I'm happy with the changes made here. I think getting the `communicator` and `formatter` grouped into an `interface` that gets passed around is a nice win – "Code that changes together should live together".
> 
> Removing some of the hardcoded duplication of player types and symbols should also make your code easier to change.
> 
> @mrfishball was the one to originally suggest a refactor so I want to make sure we get his input as well on this before merging, or if he has subsequent suggestions maybe they could be addressed in a separate PR if he agrees that these changes are an improvement.

Thank you for the review Eli 😄 and thank you a ton for all the pairing on this! 🥳 I will be sure to wait for Stevens review and approval as well!

File: test/communication/command_line/command_line_formatter_test.exs
Comment (mrfishball) [inline test/communication/command_line/command_line_formatter_test.exs:26]: I wonder if we need the first test at all. I think this test here is suffice to show that `format_board` can read empty slots as well as occupied ones. What do you think?

Comment (jonmanahan) [reply test/communication/command_line/command_line_formatter_test.exs:26]: Although I think this test is sufficient in showing the board can read empty and occupied slots, I put this starting test in there to make sure a unit test fails if you copy and paste a specific board (ie the board from the second test) into the `format_board` directly. Eli and I talked about in testing how its good to have a unit test fail when an input is "faked" ie forced to be what is in the test, so I put in two tests to make sure this happened. I hope I explained this well! If you still think it would be better to not have it I can remove it 😄, thank you for the help!

Comment (mrfishball) [reply test/communication/command_line/command_line_formatter_test.exs:26]: 👍 I also chatted with Eli about that so it's all good.

File: lib/game/tic_tac_toe.ex
Comment (mrfishball) [inline lib/game/tic_tac_toe.ex:14]: This is great, I really like the term used here.

File: test/game/game_setup_test.exs
Comment (mrfishball) [inline test/game/game_setup_test.exs:15]: I wonder if there's a way we can make the input here more explicit. Because user input determines how this function behave and affects the test outcome, it'll be easier for anyone coming into the codebase to understand what these inputs mean without having to rely on memory or past experience. For example:
```
...
user_inputs = [select_to_play_with_computer, select_to_play_with_an_easy_computer]
...
```

**Review by mrfishball — APPROVED** [review]
Looks good, thanks for doing this refactoring!

File: test/game/game_setup_test.exs
Comment (elijahsullivan) [reply test/game/game_setup_test.exs:15]: +1 @mrfishball

**Comment by jonmanahan** [comment]
> Looks good, thanks for doing this refactoring!

Thank you for the review as well Steven! 😄

File: test/game/game_setup_test.exs
Comment (mrfishball) [reply test/game/game_setup_test.exs:15]: For the record, Jon had a question about the comment I made above and I didn't get to address the question in time so I asked Jon to go ahead with the merge and made the suggested change in future PRs.

**Pull request merged**
